cpp/src/copying/pack.cpp

```diff
@@ -0,0 +1,244 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cudf/detail/copy.hpp>
+#include <cudf/detail/nvtx/ranges.hpp>
+
+#include <jit/type.h>
+
+namespace cudf {
+namespace detail {
+
+namespace {
+
+/**
+ * @brief The data that is stored as anonymous bytes in the `packed_columns` metadata
+ * field.
+ *
+ * The metadata field of the `packed_columns` struct is simply an array of these.
+ * This struct is exposed here because it is needed by both contiguous_split, pack
+ * and unpack.
+ */
+struct serialized_column {
+  data_type type;
+  size_type size;
+  size_type null_count;
+  int64_t data_offset;       // offset into contiguous data buffer, or -1 if column data is null
+  int64_t null_mask_offset;  // offset into contiguous data buffer, or -1 if column data is null
+  size_type num_children;
+  // padding explicitly so we can force initialize the value to 0, allowing us to do a byte-by-byte
+  // comparison between the output anonymized byte buffers for testing purposes.
```

Comment:
```suggestion
  // Explicitly pad to avoid uninitialized padding bits, allowing `serialized_column` to be bit-wise comparable
  // for testing purposes
```